viamonvif/service.go

```diff
@@ -131,12 +140,165 @@ func (dis *rtspDiscovery) DiscoverResources(ctx context.Context, extra map[strin
 	return cams, nil
 }
 
+func (rc *rtspDiscovery) DoCommand(ctx context.Context, command map[string]interface{}) (map[string]interface{}, error) {
+	cmd, ok := command["command"].(string)
+	if !ok {
+		return nil, errors.New("invalid command type")
+	}
+
+	switch cmd {
+	case "snapshot":
+		rc.logger.Debugf("snapshot command received")
+		snapshotReq, err := toSnapshotCommand(command)
+		if err != nil {
+			return nil, err
+		}
+
+		// Parse the URL to extract credentials
+		parsedURL, err := url.Parse(snapshotReq.snapshotURI)
+		if err != nil {
+			return nil, fmt.Errorf("invalid snapshot URI: %w", err)
+		}
+
+		// Extract credentials
+		var username, password string
+		if parsedURL.User != nil {
+			username = parsedURL.User.Username()
+			password, _ = parsedURL.User.Password()
+			rc.logger.Debugf("Using credentials: username=%s", username)
+		}
+
+		// Create HTTP client
+		client := &http.Client{
+			Timeout: 20 * time.Second,
```

Comment:
do we need to specify this timeout or can we just rely on the context being cancelled to terminate the request if the user's request times out?

Reply:
I think it would be a good idea to add a "failsafe" upper bound timeout. In case the ctx never gets canceled, I am assuming we do not want to hang forever.